invenio_rdm_records/views.py

```diff
@@ -33,6 +51,78 @@ def init(state):
     iregistry.register(ext.records_service.draft_indexer, indexer_id="records-drafts")
 
 
+@blueprint.route("/access-requests/verify")
+def verify_access_request_token():
+    """UI endpoint for verifying guest access request tokens.
+
+    When the token is verified successfully, a new guest access request will be created
+    and the token object will be deleted from the database.
+    The token value will be stored with the newly created request and grant access
+    permissions to the request details.
+    """
+    # TODO check how the email verification logic works and get inspiration from there
+    token = request.args.get("access_request_token")
+    access_request = None
+    try:
+        access_request = current_service.access.create_guest_access_request(
+            identity=g.identity, token=token
+        )
+    except DuplicateAccessRequestError as e:
+        if e.request_ids:
+            duplicate_request = current_requests_service.read(
+                identity=system_identity, id_=e.request_ids[0]
+            )
+            url = duplicate_request.links["self_html"]
+            token = duplicate_request.data["payload"]["token"]
+            return redirect(f"{url}?access_request_token={token}")
+
+    if access_request is None:
+        abort(404)
+
+    url = f"{access_request.links['self_html']}?access_request_token={token}"
+
+    send_email(
```

Comment:
I would say that we might want to use the notification system instead, so that common login (templating, or maybe header/footer, etc...) will be used.

Reply:
Sure, but then we'll need to make sure that the email notification backend is always available because that's the basis for the entire guest-based access request workflow.

Reply:
I think it would be ok to use the notification system, as we now also rely on the correct configuration of the e-mail client for sending emails. Repository admins would have to actively remove the notification backend, to make this not work. If documented accordingly, this should be fine.